public/app/features/plugins/extensions/registry/AddedFunctionsRegistry.ts

```diff
@@ -0,0 +1,74 @@
+import { ReplaySubject } from 'rxjs';
+
+import { PluginExtensionAddedFunctionConfig } from '@grafana/data';
+
+import * as errors from '../errors';
+
+import { PluginExtensionConfigs, Registry, RegistryType } from './Registry';
+
+const logPrefix = 'Could not register function extension. Reason:';
+
+export type AddedFunctionsRegistryItem = {
+  pluginId: string;
+  title: string;
+  fn: unknown;
+  description?: string;
+};
+
+export class AddedFunctionsRegistry extends Registry<AddedFunctionsRegistryItem[], PluginExtensionAddedFunctionConfig> {
+  constructor(
+    options: {
+      registrySubject?: ReplaySubject<RegistryType<AddedFunctionsRegistryItem[]>>;
+      initialState?: RegistryType<AddedFunctionsRegistryItem[]>;
+    } = {}
+  ) {
+    super(options);
+  }
+
+  mapToRegistry(
+    registry: RegistryType<AddedFunctionsRegistryItem[]>,
+    item: PluginExtensionConfigs<PluginExtensionAddedFunctionConfig>
+  ): RegistryType<AddedFunctionsRegistryItem[]> {
+    const { pluginId, configs } = item;
+    for (const config of configs) {
+      const configLog = this.logger.child({
+        title: config.title,
+        pluginId,
+      });
+
+      if (!config.title) {
+        configLog.error(`${logPrefix} ${errors.TITLE_MISSING}`);
+        continue;
+      }
```

Comment:
I guess it would make sense to validate if it is a callable (right now we can pass anything)? 
But let me know if I missed something.

For example:
```ts
if (typeof(config.fn) !== 'function') {
  configLog.error(`${logPrefix} The registered extension needs to be a function.`);
}